amundsen_application/api/mail/v0.py

```diff
@@ -83,7 +89,33 @@ def notification() -> Response:
     try:
         data = request.get_json()
         # TODO: When making a real api call, check the status code and return appropriate response
-        return make_response(jsonify({'msg': data}), HTTPStatus.OK)
+        mail_client = app.config['MAIL_CLIENT']
+
+        if not mail_client:
+            message = 'An instance of BaseMailClient client must be configured on MAIL_CLIENT'
+            logging.exception(message)
+            return make_response(jsonify({'msg': message}), HTTPStatus.NOT_IMPLEMENTED)
+
+        notification_content = get_notification_content(data['notificationType'], data['options'])
+
+        response = mail_client.send_email(
+            recipients=data['recipients'],
```

Comment:
We can add some safety here by using `get_query_param` from `request_utils`.  You can see examples of how we use it `metadata/v0`. We want to treat the React application as a black box (i.e. pretend we are not writing the logic or that we know the logic) to protect against the potential for something to go wrong there and the values don't get passed. 

If we use something like `get_query_param` (which could be renamed to something more applicable now but don't worry about that), if something goes wrong at least that method will raise an exception with a helpful error message about what parameter is missing.